src/Storage/Device/Local.php

```diff
@@ -96,12 +144,17 @@ public function upload($source, $path): bool
      *
      * @return string
      */
-    public function read(string $path): string
+    public function read(string $path, int $offset = 0, int $length = null): string
     {
         if(!$this->exists($path)) {
-            throw new Exception("File Not Found", 404);
+            throw new Exception("File Not Found");
+        }
+        if (\version_compare(PHP_VERSION, '8.0', '<')) {
```

Comment:
@eldadfux whats the targeted versions that we are trying to support. I think its fine to just deprecate PHP 7 support at this point to reduce maintenance overall 🤔

Reply:
Yes agree, let's drop 7.* support.